server/hotrod/src/main/scala/org/infinispan/server/hotrod/transport/HotRodChannelInitializer.scala

```diff
@@ -16,19 +19,24 @@ class HotRodChannelInitializer(val server: HotRodServer, transport: => NettyTran
                                val encoder: ChannelOutboundHandler, threadNamePrefix: String)
       extends NettyChannelInitializer(server, transport, encoder) {
 
+   val executor = Executors.newFixedThreadPool(server.getConfiguration.workerThreads(),
+      new DefaultThreadFactory(threadNamePrefix + "ServerHandler"))
+
    override def initChannel(ch: Channel): Unit = {
       super.initChannel(ch)
-      // Any inbound handler after this point should really be using this executionGroup
-      val executionGroup = new DefaultEventExecutorGroup(transport.configuration.workerThreads,
-         new DefaultThreadFactory(threadNamePrefix + "ServerHandler"))
-      if (server.getConfiguration.authentication().enabled()) {
-         ch.pipeline().addLast(executionGroup, "authentication", new AuthenticationHandler(server))
+      val authHandler = if (server.getConfiguration.authentication().enabled()) new AuthenticationHandler(server) else null
+      if (authHandler != null) {
+         ch.pipeline().addLast("authentication-1", authHandler)
       }
-      ch.pipeline.addLast(executionGroup, "handler", new ContextHandler(server, transport))
-      ch.pipeline.addLast(executionGroup, "exception", new HotRodExceptionHandler)
+      ch.pipeline.addLast("local-handler", new LocalContextHandler(transport))
+
+      ch.pipeline.addLast("handler", new ContextHandler(server, transport, executor))
+      ch.pipeline.addLast("exception", new HotRodExceptionHandler)
 
       // Logging handlers
-      ch.pipeline.addBefore("decoder", "logging", new HotRodAccessLoggingHandler)
-      ch.pipeline.addAfter("encoder", "logging-context", LoggingContextHandler.getInstance);
+      if (LogFactory.getLog(classOf[HotRodAccessLoggingHandler], classOf[JavaLog]).isTraceEnabled) {
```

Comment:
Unfortunately doing this means that if tracing is enabled dynamically after the server is started only new connections will have the access log.  Is this okay @tristantarrant ?  This is why I had the trace enabled checks in the handlers themselves.